transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringSubmissionQueue.java

```diff
@@ -51,124 +48,101 @@
     //these unsigned integer pointers(shared with the kernel) will be changed by the kernel
     private final long kHeadAddress;
     private final long kTailAddress;
-    private final long kRingMaskAddress;
-    private final long kRingEntriesAddress;
-    private final long fFlagsAdress;
+    //private final long fFlagsAdress;
     private final long kDroppedAddress;
     private final long arrayAddress;
 
     private final long submissionQueueArrayAddress;
 
-    private long sqeHead;
-    private long sqeTail;
+    private final int ringEntries;
+    private final int ringMask; // = ringEntries - 1
 
     private final int ringSize;
     private final long ringAddress;
     private final int ringFd;
 
-    private final ByteBuffer timeoutMemory;
     private final long timeoutMemoryAddress;
 
-    //private int sqeSubmitCounter;
+    private int head;
+    private int tail;
 
     IOUringSubmissionQueue(long kHeadAddress, long kTailAddress, long kRingMaskAddress, long kRingEntriesAddress,
                            long fFlagsAdress, long kDroppedAddress, long arrayAddress,
```

Comment:
seems like a few fields can be removed ?

Reply:
Some of these are used only from JNI during shutdown, and others I left because I wasn't sure whether we may have a use for them at some point (e.g. dropped SQE count). Probably we could trim down how many are actually needed for the shutdown though, since some are derived from others (like ringSize).

@1Jo1 wdyt?

Reply:
yeah dropped SQE count, it's probably useful for debugging or for logging, however it is very unlikely that the cqe(size=2 * sqe) is full, btw we could change cqe entry size as well at the beginning :)

Reply:
I already mentioned that, kFlag is required for SQPOLL

Reply:
ok fair enough...